streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordQueueTest.java

```diff
@@ -253,4 +276,17 @@ public void shouldDropOnNegativeTimestamp() {
 
         assertEquals(0, queue.size());
     }
+
+    static class RecordQueueTestTimestampExtractor implements TimestampExtractor {
```

Comment:
Should we rather add a `RecordQueue#partitionTime()` method for testing? Seems clumsy to go the detour via the timestamp extractor?

At the same time, we should have a test `shouldPassPartitionTimeToTimestampExtractor()` -- but this should be it's own test?

Reply:
I'm 50/50 on adding a method just for testing, although I do see your point.  I don't have a strong opinion either way.

Reply:
We do this all the time... And actually, we add it via #6694 anyway.

Reply:
Yeah, I know, and I've done it myself before, but always with some minor hesitation. But like you've pointed out, it doesn' t matter now since it's already added in the other PR.